flocker/control/test/test_protocol.py

```diff
@@ -778,6 +815,69 @@ def test_second_configuration_change_waits_for_first_acknowledgement(self):
             ),
         )
 
+    def test_second_configuration_change_suceeds_when_first_fails(self):
+        """
+        If the first update fails, we want to ensure that following updates
+        won't get stuck waiting, and will get updated.
+        """
+
+        def raise_unexpected_exception(self,
+                                       commandType=None,
+                                       *a, **kw):
+            raise Exception("I'm an unexpected exception")
+
+        agent = FakeAgent()
```

Comment:
It's a little confusing that these are constructed here and then used later on; maybe put all setup of this client in one place.